src/sage/modules/torsion_quadratic_module.py

```diff
@@ -546,32 +544,32 @@ def genus(self, signature_pair):
         r"""
         Return the genus defined by ``self`` and the ``signature_pair``.
 
-        If no such genus exists, raise a ``ValueError``.
+        If no such genus exists, raise a :class:`ValueError`.
 
         REFERENCES:
 
         [Nik1977]_ Corollary 1.9.4 and 1.16.3.
 
         EXAMPLES::
 
-            sage: L = IntegralLattice("D4").direct_sum(IntegralLattice("A2"))
-            sage: D = L.discriminant_group()
-            sage: genus = D.genus(L.signature_pair())
-            sage: genus
+            sage: L = IntegralLattice("D4").direct_sum(IntegralLattice("A2"))           # optional - sage.combinat
+            sage: D = L.discriminant_group()                                            # optional - sage.combinat
```

Comment:
When do you need `# optional - sage.combinat` tag? You don't add them for the file `sage/modules/free_quadratic_module_integer_symmetric.py`.

In general, how do you know that doctests for `IntegralLattice` needs the tag? This is not obvious at all.

Reply:
The first question is silly. `IntegralLattice` is defined in the module.

Let me change the question. Why is it `sage.combinat`?

Reply:
A preliminary inspection of `sage.combinat` is in #33660.